File: Cinema_groupe4_back/cinema_backend/src/main/java/com/intiformation/modele/Salle.java
package com.intiformation.modele;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Salle {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long idSalle;
	private int numero;
	private int nombreDePlaces;
	@ManyToOne
	@JoinColumn(name = "cinema_id", referencedColumnName = "idCinema")
	private Cinema cinema;

	public Salle() {
	}

	public Salle(int numero, int nombreDePlaces, Cinema cinema) {
		this.numero = numero;
		this.nombreDePlaces = nombreDePlaces;
		this.cinema = cinema;
	}

	public Salle(long idSalle, int numero, int nombreDePlaces, Cinema cinema) {
		this.idSalle = idSalle;
		this.numero = numero;
		this.nombreDePlaces = nombreDePlaces;
		this.cinema = cinema;
	}

	public long getIdSalle() {
		return idSalle;
	}

	public void setIdSalle(long idSalle) {
		this.idSalle = idSalle;
	}

	public int getNumero() {
		return numero;
	}

	public void setNumero(int numero) {
		this.numero = numero;
	}

	public int getNombreDePlaces() {
		return nombreDePlaces;
	}

	public void setNombreDePlaces(int nombreDePlaces) {
		this.nombreDePlaces = nombreDePlaces;
	}

	public Cinema getCinema() {
		return cinema;
	}

	public void setCinema(Cinema cinema) {
		this.cinema = cinema;
	}

}

File: Cinema_groupe4_front/cinema_front/src/app/model/salle.ts
import {Cinema} from "./cinema"

export class Salle{

    private _idSalle : number;
    private _numero : number;
    private _nombrePlace : number;
    private _cinema : Cinema
    
    constructor(idSalle : number, numero : number, nombrePlace : number, cinema : Cinema){
        this._idSalle = idSalle;
        this._numero = numero;
        this._nombrePlace = nombrePlace;
        this._cinema = cinema;
    }

    get idSalle() : number {
        return this._idSalle;
    }

    set idSalle(value : number) {
        this._idSalle = value;
    }

    get numero() : number {
        return this._numero;
    }

    set numero(value : number) {
        this._numero = value;
    }

    get nombrePlace() : number {
        return this._nombrePlace;
    }

    set nombrePlace(value : number) {
        this._nombrePlace = value;
    }

    get cinema() : Cinema {
        return this._cinema;
    }

    set cinema(value : Cinema) {
        this._cinema = value;
    }


}